lib/liquid/standardfilters.rb

```diff
@@ -488,16 +832,18 @@ def at_most(input, n)
       result.is_a?(BigDecimal) ? result.to_f : result
     end
 
-    # Set a default value when the input is nil, false or empty
-    #
-    # Example:
-    #    {{ product.title | default: "No Title" }}
-    #
-    # Use `allow_false` when an input should only be tested against nil or empty and not false.
-    #
-    # Example:
-    #    {{ product.title | default: "No Title", allow_false: true }}
+    # @liquid_public_docs
+    # @liquid_type filter
+    # @liquid_category array
+    # @liquid_summary
+    #   Sets a default value for any variable whose value is one of the following:
     #
+    #   - [`EmptyDrop`](/api/liquid/basics#emptydrop)
+    #   - [`false`](/api/liquid/basics#truthy-and-falsy)
+    #   - [`nil`](/api/liquid/basics#nil)
+    # @liquid_syntax variable | default: variable
+    # @liquid_optional_param allow_false [boolean] Whether to use false values instead of the default.
+    # @liquid_return [variable]
```

Comment:
I think we should just not specify a return type here, as it's determined by how you use the filter.